properties/base/base.py

```diff
@@ -510,36 +514,52 @@ def deserialize(cls, value, trusted=False, strict=False,                   #pyli
         """
         if not isinstance(value, dict):
             raise ValueError('HasProperties must deserialize from dictionary')
-        input_class = value.get('__class__')
-        if not input_class or input_class == cls.__name__:
-            pass
-        elif trusted and input_class in cls._REGISTRY:
-            cls = cls._REGISTRY[input_class]
-        elif strict:
-            raise utils.ValidationError(
-                'Class name {} from input dictionary does not match input '
-                'class {}'.format(input_class, cls.__name__))
-        kwargs.update({'trusted': trusted, 'strict': strict})
+        cls = cls._deserialize_class(value.get('__class__'), trusted, strict)
+        instance = kwargs.pop('_instance', None)
+        if instance is not None and not isinstance(instance, cls):
+            raise ValueError(
+                'Input _instance must be of class {}, not {}'.format(
+                    cls.__name__, instance.__class__.__name__
+                )
+            )
         state, unused = utils.filter_props(cls, value, True)
         unused.pop('__class__', None)
         if unused and strict:
-            raise utils.ValidationError(
+            raise ValueError(
```

Comment:
What's the thinking on this change?

Reply:
I think `ValidationError`s should only be used with `validate` - this is just a bad input dictionary. Possibly we want a `DeserializeError` or something?

Precedent is set up here https://github.com/seequent/properties/pull/234/files/ff253ea588fb31ca91ba36732bde53f862c4c029#diff-55e8c5dfe4b1346a91ff48e34999597cR516 with just the value error.

For what it's worth, this is not backwards-incompatible with master...

Reply:
Right, okay, that is fair